pkg/dns/serviceresolver.go

```diff
@@ -48,16 +47,39 @@ func NewServiceResolver(config *server.Config, accessor ServiceAccessor, endpoin
 
 // Records implements the SkyDNS Backend interface and returns standard records for
 // a name.
+//
+// The standard pattern is <prefix>.<service_name>.<namespace>.(svc|endpoints).<base>
+//
+// * prefix may be any series of prefix values
+// * service_name and namespace must locate a real service
+// * svc indicates standard service rules apply (portalIP or endpoints as A records)
+//   * reverse lookup of IP is only possible for portalIP
+//   * SRV records are returned for each host+port combination as:
+//     _<port_name>._<port_protocol>.<dns>
+//     _<port_name>.<endpoint_id>.<dns>
+//   * endpoint_id is "portal" when portalIP is set
+// * endpoints always returns each individual endpoint as A records
```

Comment:
I see.  Is this really needed?  Brian convinced me not to add it just because I could.